completions/ssh

```diff
@@ -512,15 +512,15 @@ _comp_xfunc_scp_compgen_remote_files()
         # shellcheck disable=SC2090
         _files=$(ssh -o 'Batchmode yes' "$_userhost" \
             command ls -aF1dL "$_path*" 2>/dev/null |
-            command sed -e 's/'"$_comp_cmd_scp__path_esc"'/'"$_escape_replacement"'/g' -e '/[^\/]$/d')
+            command sed -e 's/'"$_comp_cmd_scp__path_esc"'/'"$_escape_replacement"'/g' -e '/[^/]$/d')
     else
         # escape problematic characters; remove executables, aliases, pipes
         # and sockets; add space at end of file names
         # shellcheck disable=SC2090
         _files=$(ssh -o 'Batchmode yes' "$_userhost" \
             command ls -aF1dL "$_path*" 2>/dev/null |
             command sed -e 's/'"$_comp_cmd_scp__path_esc"'/'"$_escape_replacement"'/g' -e 's/[*@|=]$//g' \
-                -e 's/[^\/]$/& /g')
+                -e 's/[^/]$/& /g')
```

Comment:
I tried to add a test for this `if`-branch, but it seems difficult to test it within the current test framework. This branch tries to append a space when it is not a directory, but `bash-completion`'s test framework removes the trailing whitespace when it extracts the generated completions. As a result, we cannot detect whether a suffix space is added by the completion function or not.

Reply:
I realized that the raw value is actually stored in `self.output` in `class CompletionResult`, and it is used in existing tests.

```console
$ git grep 'completion\.output'
test/t/test_chown.py:        assert completion.output.endswith(" ")
test/t/test_chown.py:        assert completion.output.endswith(" ")
test/t/test_chown.py:        assert completion.output.endswith(" ")
test/t/test_mkdir.py:        assert completion.output == "foo"
test/t/test_mkdir.py:        assert completion == [completion.output]
```

Reply:
> I realized that the raw value is actually stored in `self.output` in `class CompletionResult`, and it is used in existing tests.
> 
> ```console
> $ git grep 'completion\.output'
> test/t/test_chown.py:        assert completion.output.endswith(" ")
> test/t/test_chown.py:        assert completion.output.endswith(" ")
> test/t/test_chown.py:        assert completion.output.endswith(" ")
> test/t/test_mkdir.py:        assert completion.output == "foo"
> test/t/test_mkdir.py:        assert completion == [completion.output]
> ```

This is addressed in #1408.